src/protocol/mod.rs

```diff
@@ -243,3 +244,15 @@ impl From<RecordId> for u32 {
         v.0
     }
 }
+
+impl From<RecordId> for usize {
+    fn from(r: RecordId) -> Self {
+        r.0 as usize
+    }
+}
+
+impl AddAssign<usize> for RecordId {
```

Comment:
i saw you're doing `*offset += 1`, but would it be better to `impl AddAssign<u32> for RecordId` and change that line to `*offset += 1u32`?

Reply:
just curious why it might be better - you need to type more for something that is already convertible to `usize`?

Reply:
only to avoid the conversion + `unwrap`. seems like that is unnecessary work